### DISCUSSION TITLE
Disable update checker for F-Droid

### DISCUSSION THREAD

**Discussion by shuvashish76** [TeamNewPipe/NewPipe#10785] [Questions] [answered]
- The built **signed by F-Droid** - Update checker feature removed completely
![59a8a723-4336-46d5-868e-fd1968baed52](https://github.com/TeamNewPipe/NewPipe/assets/48226312/42b77171-bce9-4b9a-8fc3-6cad8b30aece)

- The built **signed by original developer** - Update checker enabled by default
![51e71262-907e-4f88-bf08-111c38770a48](https://github.com/TeamNewPipe/NewPipe/assets/48226312/6ca64123-256f-4bd7-aa64-c2441cb66c96)
![c756e070-1abb-4485-a4ed-a78a759de1fe](https://github.com/TeamNewPipe/NewPipe/assets/48226312/f254c009-4b3f-464c-951a-2ea358cff4fb)

As per their criteria It would require a [Tracking](https://f-droid.org/en/docs/Anti-Features/#Tracking) Anti-Feature tag if enabled by default.

```
- Checking for updates without your knowledge or permission
Examples of where it would not be applied - any of the above, if the functionality is opt-in (i.e. you are asked before it happens) and disabled by default.

```

Request to disable update checker for the _built signed by original developer_ as well.
F-Droid issue : https://gitlab.com/fdroid/fdroiddata/-/issues/3147

**Comment by licaon-kter** [comment]
The buzzword here is _"opt-in"_, so a simple pop-up on (first) install _"Hey, do you want to check for updates straight from the NewPipe Team? Y/N"_ can be a solution.


@shuvashish76 
> Request to disable update checker for the built signed by original developer as well.

Do note that, while the builds on F-Droid are signed one by F-Droid and one by NewPipe, the APK is the same (since it's reproducible)

**Reply by licaon-kter** [reply]
Now I'm confused, if I install each APK indeed the F-Droid signed one does not show the updater, **the app checks the signature or what magic is it used?** :)

**Reply by licaon-kter** [reply]
Ok, yup https://github.com/TeamNewPipe/NewPipe/blob/v0.25.2/app/src/main/java/org/schabi/newpipe/util/ReleaseVersionUtil.kt#L22-L27

**Comment by licaon-kter** [comment]
The gist of it is this
> The software must not download additional executable binary files (e.g. addons, auto-updates, etc.) **without explicit user consent. Consent means it needs to be opt-in** (it must not be harder to decline than to accept or presented in a way users are likely to press accept without reading) and structured in a way that **clearly explains to users that they’re choosing to bypass F-Droid’s checks if they activate it**.

Per https://f-droid.org/docs/Inclusion_Policy/

Regarding "bypass F-Droid", I've pointed [above](https://github.com/TeamNewPipe/NewPipe/discussions/10785#discussioncomment-8230436) how this can be formulated so the same APK shows an universal message to all users (be it from Github, Newpipe repo, F-Droid repo)

**Reply by shuvashish76** [reply]
Yes the inclusion policy is strictly applicable to **downloaders** but in this case NewPipe has **update checker** only hence AF.
Would be nice have to opt-in i.e. disabled by default.

**Comment by TobiGr** [comment]
Thanks for making us aware. We should have considered asking for the users' consent. I implemented a proof of concept in #10790

**Comment by Stypox** [comment, marked as answer]
@licaon-kter @shuvashish76 thank you for reporting this. Here an APK that fixes the issue, let us know if its behavior is ok: [NewPipe updatechecker.zip](https://github.com/TeamNewPipe/NewPipe/files/14802832/NewPipe.updatechecker.zip)
This is the expected behavior:
- The **first** time the app is opened, no update check is run, and no consent is asked (this is to avoid a bad onboarding and welcoming experience)
- The **second** time, no update check is run, but the user is asked for consent
- If the user accepted to check for updates, from the **third** time onwards, the app will check for updates, otherwise it will never do so

<sub>The APK is signed with @Stypox keys. The fingerprint for which to check for updates has been set to `f3c6b6ddf4f4c9e2fa5c5e65799dc388d17eae626acfa6e29f571b3520883b75`, i.e. the fingerprint of my key. The version code of the APK is that of 0.26.0 (995) instead of 0.26.1 (996) so that version checks show a notification. The package name is `org.schabi.newpipe.updatechecker`.</sub>

**Reply by shuvashish76** [reply]
Perfect, looks good to me unless F-Droid maintainers disagree. Thanks.
